File: 3_Recursion/Fundamentals of recursion/3_Sum of N Natural Numbers/Solution.c
#include <stdio.h>

int sum(int n) {
    printf("Sum(%d) is called\n", n);
    if (n == 1) {
        printf("Base Condition reached\n");
        return 1;
    }
    int sumN1 = sum(n - 1);
    printf("Sum(%d) = %d + Sum(%d)\n", n, n, n - 1);
    int sumN = n + sumN1;
    printf("Sum(%d) = %d + %d\n", n, n, sumN1);
    printf("Sum(%d) is returning %d\n", n, sumN);
    return sumN;
}

int main() {
    printf("Sum of first 5 natural numbers is %d\n", sum(5));
    return 0;
}

File: 3_Recursion/Fundamentals of recursion/3_Sum of N Natural Numbers/Solution.py
def Sum(n):
    print(f"Sum({n}) is called")
    if n==1:
        print("Base Condition reached")
        return 1
    sumN1 = Sum(n-1)
    print(f"Sum({n}) = {n} + Sum({n-1})")
    
    sumN = n + sumN1
    print(f"Sum({n}) = {n} + {sumN1}")
    
    print(f"Sum({n}) is returning {sumN}")
    return sumN
    
print("Sum of first 5 natural numbers is", Sum(5))